expan/core/util.py

```diff
@@ -1,172 +1,117 @@
-from warnings import warn
+import json
+import logging
+from enum import Enum
 
 import numpy as np
 import pandas as pd
 from datetime import datetime, timedelta
 
+logger = logging.getLogger(__name__)
 
-def find_list_of_dicts_element(items, key1, value, key2):
-    return [item[key2] for item in items if item[key1] == value][0]
 
-def is_number_and_nan(obj):
-    return obj != obj
-
-def get_column_names_by_type(df, dtype):
-    return [c for c in df.columns if np.issubdtype(df.dtypes[c], dtype)]
+class JsonSerializable(object):
+    """ Interface for serializable classes."""
+    def toJson(self):
+        return json.dumps(self, default=lambda o: o.name if isinstance(o, Enum) else o.__dict__, sort_keys=True, indent=4)
 
-def drop_nan(np_array):
-    if np_array.ndim == 1:
-        return np_array[~np.isnan(np_array)]
-    elif np_array.ndim == 2:
-        return np_array[~np.isnan(np_array).any(axis=1)]
+    def __repr__(self):
+        return self.toJson()
 
 
-def scale_range(x, new_min=0.0, new_max=1.0, old_min=None, old_max=None, squash_outside_range=True, squash_inf=False, ):
+def find_value_by_key_with_condition(items, condition_key, condition_value, lookup_key):
+    """ Find the value of lookup key where the dictionary contains condition key = condition value.
+    
+    :param items: list of dictionaries
+    :type  items: list
+    :param condition_key: condition key
+    :param condition_value: condition value
+    :param lookup_key: lookup key
+    
+    :return: lookup value
     """
-    Scales a sequence to fit within a new range.
-
-    If squash_inf is set, then infinite values will take on the
-    extremes of the new range (as opposed to staying infinite).
-
-    Args:
-        x:
-        new_min:
-        new_max:
-        old_min:
-        old_max:
-        squash_outside_range:
-        squash_inf:
-
-    Note:
-        Infinity in the input is disregarded in the construction of the scale of the mapping.
-
-  >>> scale_range([1,3,5])
-  array([ 0. ,  0.5,  1. ])
-
-  >>> scale_range([1,2,3,4,5])
-  array([ 0.  ,  0.25,  0.5 ,  0.75,  1.  ])
-
-  >>> scale_range([1,3,5, np.inf])
-  array([ 0. ,  0.5,  1. ,  inf])
-
-  >>> scale_range([1,3,5, -np.inf])
-  array([ 0. ,  0.5,  1. , -inf])
-
-  >>> scale_range([1,3,5, -np.inf], squash_inf=True)
-  array([ 0. ,  0.5,  1. ,  0. ])
-
-  >>> scale_range([1,3,5, np.inf], squash_inf=True)
-  array([ 0. ,  0.5,  1. ,  1. ])
-
-  >>> scale_range([1,3,5], new_min=0.5)
-  array([ 0.5 ,  0.75,  1.  ])
-
-  >>> scale_range([1,3,5], old_min=1, old_max=4)
-  array([ 0.        ,  0.66666667,  1.        ])
-
-  >>> scale_range([5], old_max=4)
-  array([ 1.])
+    return [item[lookup_key] for item in items if item[condition_key] == condition_value][0]
```

Comment:
this one is prone to throwing `KeyError`s

Reply:
Yes. we have created new units tests in ```test_util.py``` which covers this case.